### PR TITLE
feat(hook): Add sentry pride logo hook

### PR DESCRIPTION
**Pull request by priscilawebdev**
Add a hook for the Sentry pride logo. 

The Sentry pride logo lives in `getsentry` and it is only visible if:

- The organization is not self-hosted and 
- The feature flag `organizations: sentry-pride-logo-footer` is enabled 

for now the sentry pride logo will be only shown in the footer of our SaaS application

Requires https://github.com/getsentry/getsentry/pull/10655

### CODE REVIEW

File: static/app/components/footer.tsx [hunk 1]
```diff
@@ -39,7 +47,14 @@ function BaseFooter({className}: Props) {
...
       </LeftLinks>
-      <LogoLink />
+      <SentryLogoLink href="https://sentry.io/welcome/" tabIndex={-1}>
+        {!isSelfHosted &&
+        !!(organization.features ?? []).includes('sentry-pride-logo-footer') ? (
+          <SentryPrideLogoHook size="lg" />
+        ) : (
+          <IconSentry size="lg" />
+        )}
+      </SentryLogoLink>
```
Comment (matejminar) [inline static/app/components/footer.tsx:57]:
```suggestion
      <SentryLogoLink href="https://sentry.io/welcome/" tabIndex={-1}>
          <SentryLogoHook size="lg" />
      </SentryLogoLink>
```
No need to check isSelfHosted as the hook will default to the regular logo. I would move the feature flag check to getsentry.

Comment (priscilawebdev) [reply static/app/components/footer.tsx]: can confirm that we don't need to tch `isSelfHosted` ..thanks Matej!

File: static/app/components/footer.tsx [hunk 2]
```diff
@@ -17,6 +24,7 @@ type Props = {
 function BaseFooter({className}: Props) {
   const {isSelfHosted, version, privacyUrl, termsUrl, demoMode} =
     useLegacyStore(ConfigStore);
+  const organization = useOrganization();
```
[the remark this answers is not in the document]
Comment (evanpurkhiser) [reply static/app/components/footer.tsx:27]: https://github.com/getsentry/sentry/pull/49840

[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/footer.tsx]: will have to update the code according to your PR afterward, but we can merge this PR meanwhile :)

File: static/app/components/footer.tsx [hunk 3]
```diff
@@ -2,13 +2,20 @@ import {Fragment} from 'react';
...
 import ConfigStore from 'sentry/stores/configStore';
 import {useLegacyStore} from 'sentry/stores/useLegacyStore';
 import {space} from 'sentry/styles/space';
 import getDynamicText from 'sentry/utils/getDynamicText';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const SentryPrideLogoHook = HookOrDefault({
+  hookName: 'component:sentry-pride-logo',
+  defaultComponent: () => <IconSentry size="lg" />,
+});
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/footer.tsx:18]: sure, but it is just seems weird that we have both `IconSentryLogo` and `SentryLogoHook`

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**